pkg/controller/operators/catalog/operator.go

```diff
@@ -1654,18 +1686,70 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 				plan.Status.Plan[i].Status = status
 
 			default:
-				return v1alpha1.ErrInvalidInstallPlan
-			}
+				if !isSupported(step.Resource.Kind) {
+					// Not a supported resource
+					plan.Status.Plan[i].Status = v1alpha1.StepStatusUnsupportedResource
+					return v1alpha1.ErrInvalidInstallPlan
+				}
+
+				// Entering unstructured territory... To infinity and beyond!
```

Comment:
This comment is unnecessary and probably should be removed unless you add some information here such as "Install other supported resources". You don't really install unstructured resources. You simply decode yaml into unstructured object in order to install it.

Reply:
Seems to be a copy of https://github.com/operator-framework/operator-lifecycle-manager/pull/1227#discussion_r369935743